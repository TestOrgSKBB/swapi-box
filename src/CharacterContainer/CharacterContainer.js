import React from 'react';
import { NavLink } from 'react-router-dom';
import $ from 'jquery';
import CharacterCard from '../CharacterCard/CharacterCard';

const CharacterContainer = ( props ) => {
  $('.title').text('Characters');
  const displayCharacters = props.characters.map((character, i) => <CharacterCard character={character} updateFavorite={props.updateFavorite} key={i} />);
  const opening_crawl = props.opening_crawl;

  return (
    <>
      <NavLink to='/movies/favorite-characters' >
        <button>View Favorites ({props.numFavorites})</button>
      </NavLink>
      {displayCharacters}
      <h1>{opening_crawl}</h1>
    </>
  );
 };  


export default CharacterContainer;