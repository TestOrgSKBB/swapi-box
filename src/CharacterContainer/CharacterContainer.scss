.section__characters {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  font-family: 'Pathway Gothic One', sans-serif;
  overflow: scroll;
  height: 84vh;
  border: 2px solid purple;
  // margin-top: 10%;
}

.div__crawl {
  position: relative;
  top: -100px;
  transform-origin: 50% 100%;
  animation: crawl 60s infinite linear;
  z-index: -1;
}

@keyframes crawl {
  0% {
    top: 0;
    transform: rotateX(20deg) translateZ(0);
  }
  100% { 
    top: -6000px;
    transform: rotateX(25deg) translateZ(-2500px);
  }
}

.star-wars {
  display: flex;
  justify-content: center;
  perspective: 400px;
  color: #feda4a;
  font-family: 'Pathway Gothic One', sans-serif;
  font-size: 500%;
  font-weight: 600;
  letter-spacing: 6px;
  line-height: 150%;
  text-align: justify;
  height: 68%;
  z-index: -10;
  position: absolute
}
