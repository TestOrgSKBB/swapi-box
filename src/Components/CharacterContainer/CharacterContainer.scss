.section__characters {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  font-family: 'Pathway Gothic One', sans-serif;
  overflow: scroll;
  height: 75vh;
}

.div__crawl {
  position: relative;
  top: -100px;
  transform-origin: 50% 100%;
  animation: crawl 60s infinite linear;
  z-index: -1;
}

@keyframes crawl {
  0% {
    top: 0;
    transform: rotateX(20deg) translateZ(0);
  }
  100% { 
    top: -6000px;
    transform: rotateX(25deg) translateZ(-2500px);
  }
}

.star-wars {
  display: flex;
  justify-content: center;
  perspective: 400px;
  color: #feda4a;
  font-family: 'Pathway Gothic One', sans-serif;
  font-size: 500%;
  font-weight: 600;
  letter-spacing: 6px;
  line-height: 150%;
  text-align: justify;
  height: 68%;
  z-index: -10;
  position: absolute
}

.button__view-favorites {
  background: rgba(55, 51, 62, 0.91);
  border-radius: 16em;
  border: 1px solid #919192;
  color: #919192;
  font-size: 1em;
  height: 5vh;
  justify-self: center;
  margin-left: 25%;
  width: 50%;
}

.button__view-favorites:hover {
  background: rgba(143, 171, 172, .7);
  border: 2px dotted rgba(32, 37, 40, .8);
  color: rgba(32, 37, 40, .8);
  font-weight: bold;
  cursor: pointer;
}