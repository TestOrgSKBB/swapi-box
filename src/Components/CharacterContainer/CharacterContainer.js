import React from 'react';
import { Link } from 'react-router-dom';
import CharacterCard from '../CharacterCard/CharacterCard';
import '../CharacterContainer/CharacterContainer.scss';
import PropTypes from 'prop-types';
import backImage from '../../images/back.png'

const CharacterContainer = ( props ) => {
  // $('.title').text('Characters');
  const displayCharacters = props.characters.slice(0, 10).map((character, i) => 
    <CharacterCard 
      character={character} 
      updateFavorite={props.updateFavorite} 
      key={i} 
    />);

  const opening_crawl = props.opening_crawl;

  return (
    <>
      <h1 className="character-title">Characters</h1>
      <Link to='/movies'>
        <img className="button__back" src={backImage} alt="back button"/>
      </Link>
      <Link to='/movies/favorite-characters' >
        <button className="button__view-favorites">View Favorites ({props.numFavorites})</button>
      </Link>
      <section className="section__characters">
        {displayCharacters}
        <div className="star-wars">
          <div className="div__crawl">
          <p>{opening_crawl}</p>
          </div>
        </div>
      </section>
    </>
  );
 }; 

CharacterContainer.propTypes = {
  characters: PropTypes.array,
  numFavorites: PropTypes.number,
  opening_crawl: PropTypes.string,
  updateFavorite: PropTypes.func,
  name: PropTypes.string,
  species: PropTypes.string,
  homeworld: PropTypes.string,
  handleFavorite: PropTypes.func
 };

export default CharacterContainer;