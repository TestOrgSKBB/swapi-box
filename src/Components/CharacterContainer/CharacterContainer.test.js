  
import React from 'react';
import CharacterContainer from './CharacterContainer';
import { shallow } from 'enzyme';

describe('CharacterContainer', () => {
  let wrapper;
  const characters = [
    {
      films:['A Film', 'Another Film', 'And another'],
      homeworld: 'Earth',
      name: 'A name',
      population: '2000',
      species:'human'
    },
    {
      films:['A Film 2', 'Another Film 2', 'And another 2'],
      homeworld: 'Earth 2',
      name: 'A name 2',
      population: '2000 2',
      species:'human 2'
    },
    {
      films:['A Film 3', 'Another Film 3', 'And another 3'],
      homeworld: 'Earth 3',
      name: 'A name 3',
      population: '2000 3',
      species:'human 3'
    }
  ];

 beforeEach(() => {
   wrapper = shallow(<CharacterContainer
    characters={characters}
  />);
 });

 it('should match the snapshot with the correct data', () => {
   expect(wrapper).toMatchSnapshot();
 });
 
});