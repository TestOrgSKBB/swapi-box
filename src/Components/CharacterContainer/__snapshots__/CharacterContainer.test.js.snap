// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterContainer should match the snapshot with the correct data 1`] = `
<Fragment>
  <h1
    className="character-title"
  >
    Characters
  </h1>
  <Link
    to="/movies"
  >
    <img
      alt="back button"
      className="button__back"
      src="back.png"
    />
  </Link>
  <Link
    to="/movies/favorite-characters"
  >
    <button
      className="button__view-favorites"
    >
      View Favorites (
      )
    </button>
  </Link>
  <section
    className="section__characters"
  >
    <Character
      character={
        Object {
          "films": Array [
            "A Film",
            "Another Film",
            "And another",
          ],
          "homeworld": "Earth",
          "name": "A name",
          "population": "2000",
          "species": "human",
        }
      }
      key="0"
    />
    <Character
      character={
        Object {
          "films": Array [
            "A Film 2",
            "Another Film 2",
            "And another 2",
          ],
          "homeworld": "Earth 2",
          "name": "A name 2",
          "population": "2000 2",
          "species": "human 2",
        }
      }
      key="1"
    />
    <Character
      character={
        Object {
          "films": Array [
            "A Film 3",
            "Another Film 3",
            "And another 3",
          ],
          "homeworld": "Earth 3",
          "name": "A name 3",
          "population": "2000 3",
          "species": "human 3",
        }
      }
      key="2"
    />
    <div
      className="star-wars"
    >
      <div
        className="div__crawl"
      >
        <p />
      </div>
    </div>
  </section>
</Fragment>
`;
